case_studies/weak_lensing/lensing_encoder.py

```diff
@@ -0,0 +1,172 @@
+import sys
+from typing import Optional
+
+import torch
+from torch.optim import Adam
+from torch.optim.lr_scheduler import MultiStepLR
+from torchmetrics import MetricCollection
+
+from bliss.catalog import TileCatalog
+from bliss.encoder.encoder import Encoder
+from bliss.encoder.variational_dist import VariationalDist
+from bliss.global_env import GlobalEnv
+from case_studies.weak_lensing.lensing_convnet import WeakLensingCatalogNet, WeakLensingFeaturesNet
+
+
+class WeakLensingEncoder(Encoder):
+    def __init__(
+        self,
+        survey_bands: list,
+        tile_slen: int,
+        image_normalizers: list,
+        var_dist: VariationalDist,
+        sample_image_renders: MetricCollection,
+        mode_metrics: MetricCollection,
+        sample_metrics: Optional[MetricCollection] = None,
+        optimizer_params: Optional[dict] = None,
+        scheduler_params: Optional[dict] = None,
+        reference_band: int = 2,
+        **kwargs,
+    ):
+        super().__init__(
```

Comment:
Same here, how about passing `*args` and `**kwargs` to the parent class? (and removing named arguments in the construct that are just passed on to `Encoder`)

Reply:
will remove named arguments, but same issue as above with hardcoding certain params

Reply:
Ok sounds good